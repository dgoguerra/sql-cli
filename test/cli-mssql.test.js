const Knex = require("knex");
const { cliTestSuite } = require("./cli-test-suite");

const TEST_MSSQL_CONN = {
  server: "127.0.0.1",
  port: 21433,
  user: "test",
  password: "Secret123",
  database: "test_db",
  options: { enableArithAbort: true },
};

cliTestSuite("mssql", () =>
  Knex({ client: "mssql", connection: TEST_MSSQL_CONN })
);
