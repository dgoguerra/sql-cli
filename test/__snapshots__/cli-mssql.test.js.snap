// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI test: mssql can create conn aliases 1`] = `
"alias  conn                                     
alias1 mysql://app:secret@127.0.0.1:1234/dbname 
alias2 sqlite3:///path/to/file.db               
"
`;

exports[`CLI test: mssql can delete conn aliases 1`] = `
"alias  conn                       
alias2 sqlite3:///path/to/file.db 
"
`;

exports[`CLI test: mssql can diff schemas 1`] = `
"No tables with changes: 3x similar (hidden)
"
`;

exports[`CLI test: mssql can diff tables 1`] = `
"column  type                   
field_1 int nullable → decimal 

index                                                         unique columns 
PK__table_1__0000000000000000 → PK__table_2__0000000000000000 true   id      

Columns: 1x changed, 4x similar (hidden)
Indexes: 1x changed
"
`;

exports[`CLI test: mssql can diff tables data 1`] = `
"Diff of tables content (first 20 rows):

id field_1    updated_at                                          
1  12 → 12.3  [null] → 2020-07-24T18:34:00.000Z                   
2  30 → 30.45 2020-07-24T18:34:00.000Z → 2020-07-24T19:25:00.000Z 
"
`;

exports[`CLI test: mssql can list tables 1`] = `
"table   rows bytes 
table_1 2    16 kB 
table_2 2    16 kB 
table_3 0    0 B   

(32 kB in 3 tables)
"
`;

exports[`CLI test: mssql can load database dump 1`] = `
"column     type         nullable 
id         int          false    
field_1    int          true     
field_2    nvarchar(-1) true     
created_at datetime2    true     
updated_at datetime2    true     

index                         unique columns 
PK__table_1__0000000000000000 true   id      
"
`;

exports[`CLI test: mssql can load database dump 2`] = `
"column     type         nullable 
id         int          false    
field_1    decimal      false    
field_2    nvarchar(-1) true     
created_at datetime2    true     
updated_at datetime2    true     

index                         unique columns 
PK__table_2__0000000000000000 true   id      
"
`;

exports[`CLI test: mssql can load database dump 3`] = `
"column     type      nullable 
idField    bigint    false    
field_1    bigint    true     
created_at datetime2 true     
updated_at datetime2 true     

index                         unique columns 
PK__table_3__0000000000000000 true   idField 
"
`;

exports[`CLI test: mssql can show table 1`] = `
"column     type         nullable 
id         int          false    
field_1    decimal      false    
field_2    nvarchar(-1) true     
created_at datetime2    true     
updated_at datetime2    true     

index                         unique columns 
PK__table_2__0000000000000000 true   id      
"
`;

exports[`CLI test: mssql can use a created alias 1`] = `
"table   rows bytes 
table_1 2    16 kB 
table_2 2    16 kB 
table_3 0    0 B   

(32 kB in 3 tables)
"
`;

exports[`CLI test: mssql database dump contents are valid 1`] = `
"module.exports.up = async (knex) => {
  await knex.schema.createTable(\\"table_1\\", (t) => {
    t.increments(\\"id\\");
    t.integer(\\"field_1\\");
    t.text(\\"field_2\\");
    t.timestamp(\\"created_at\\");
    t.timestamp(\\"updated_at\\");
  });
};
module.exports.down = async (knex) => {
  await knex.schema.dropTableIfExists(\\"table_1\\");
};
"
`;

exports[`CLI test: mssql database dump contents are valid 2`] = `
"module.exports.up = async (knex) => {
  await knex.schema.createTable(\\"table_2\\", (t) => {
    t.increments(\\"id\\");
    t.decimal(\\"field_1\\").notNullable();
    t.text(\\"field_2\\").defaultTo(\\"default text\\");
    t.timestamp(\\"created_at\\");
    t.timestamp(\\"updated_at\\");
  });
};
module.exports.down = async (knex) => {
  await knex.schema.dropTableIfExists(\\"table_2\\");
};
"
`;

exports[`CLI test: mssql database dump contents are valid 3`] = `
"module.exports.up = async (knex) => {
  await knex.schema.createTable(\\"table_3\\", (t) => {
    t.bigIncrements(\\"idField\\");
    t.bigInteger(\\"field_1\\");
    t.timestamp(\\"created_at\\");
    t.timestamp(\\"updated_at\\");
  });
};
module.exports.down = async (knex) => {
  await knex.schema.dropTableIfExists(\\"table_3\\");
};
"
`;

exports[`CLI test: mssql database dump contents are valid 4`] = `
"{\\"id\\":1,\\"field_1\\":12,\\"field_2\\":\\"foo\\",\\"created_at\\":\\"2020-07-24T18:34:00.000Z\\"}
{\\"id\\":2,\\"field_1\\":30,\\"field_2\\":\\"bar\\",\\"created_at\\":\\"2020-07-24T18:34:00.000Z\\",\\"updated_at\\":\\"2020-07-24T18:34:00.000Z\\"}
"
`;

exports[`CLI test: mssql database dump contents are valid 5`] = `
"{\\"id\\":1,\\"field_1\\":12.3,\\"field_2\\":\\"foo\\",\\"created_at\\":\\"2020-07-24T18:34:00.000Z\\",\\"updated_at\\":\\"2020-07-24T18:34:00.000Z\\"}
{\\"id\\":2,\\"field_1\\":30.45,\\"field_2\\":\\"bar\\",\\"created_at\\":\\"2020-07-24T18:34:00.000Z\\",\\"updated_at\\":\\"2020-07-24T19:25:00.000Z\\"}
"
`;
